### PR TITLE
[user] Add unit tests for hooks

### PR DESCRIPTION
**Pull request by jaylees14**
Add unit tests for hooks

### CODE REVIEW

File: user/user_test.go [hunk 1]
```diff
@@ -166,6 +167,98 @@ func TestCreateUserHandlerFailsOnEmptyJWT(t *testing.T) {
...
+		t.Errorf("Wrong status code: %v", res.Code)
+	}
+}
+
+// Test that an after create hook is successfully invoked
+func TestCreateUserHandlerAfterHookSucceeds(t *testing.T) {
+	mockEnv := makeMockEnv()
+
+	mockEnv.hook.AfterCreate(func(env *env, user *dao.User) *HookError {
+		// Set all response fields to Lewis
```
Comment (Lewiky) [inline user/user_test.go:221]: :)

File: user/user_test.go [hunk 2]
```diff
@@ -483,3 +793,110 @@ func TestDeleteUserHandlerFailsOnDifferentJWT(t *testing.T) {
...
+	}
+}
+
+// Test that an after delete hook is successfully invoked
+func TestDeleteUserHandlerAfterHookSucceeds(t *testing.T) {
+	mockEnv := makeMockEnv()
+	isHookExecuted := false
+
+	mockEnv.hook.AfterDelete(func(env *env) *HookError {
+		isHookExecuted = true
```
Comment (Lewiky) [inline user/user_test.go:854]: How does this work?

**Review by Lewiky — APPROVED** [review]
LGTM bruh

File: user/user_test.go (hunk 2 above)
Comment (jaylees14) [reply user/user_test.go:854]: Much closure magic

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀

**Pull request merged**